strings/base_com_ptr.h

```diff
@@ -133,8 +133,17 @@ WINRT_EXPORT namespace winrt
         template <typename To>
         bool try_as(To& to) const noexcept
         {
-            to = try_as<impl::wrapped_type_t<To>>();
-            return static_cast<bool>(to);
+            if constexpr (impl::is_com_interface_v<To> || !std::is_same_v<To, impl::wrapped_type_t<To>>)
```

Comment:
The tricky part is these `enable_if_t`'s and `if constexpr`'s. I'm still not confident that I got them all right.

Reply:
Exhaustive testing of edge cases is the only way I feel confident about this stuff.